### ISSUE TITLE
Werf cannot initialize submodules with relative url-s when non "origin" remote used

### ISSUE THREAD

**Issue by distorhead** [werf/werf#3223]
1. Suppose we are using `gitlab` remote instead of default hardcoded into git `origin`.
 2. Suppose we have added a submodule with relative url.
 3. Werf build command will try to bootstrap a separate worktree with initialized submodules to perform internal operations on git. During this bootstrap werf calls git-submodule-update command on service worktree with detached commit state. Git cannot determine default-remote according to the official docs:

```
<repository> is the URL of the new submodule’s origin repository. This may be either an absolute URL, or (if it begins with ./ or ../), the location relative to the superproject’s default remote repository (Please note that to specify a repository foo.git which is located right next to a superproject bar.git, you’ll have to use ../foo.git instead of ./foo.git - as one might expect when following the rules for relative URLs - because the evaluation of relative URLs in Git is identical to that of relative directories).

The default remote is the remote of the remote-tracking branch of the current branch. If no such remote-tracking branch exists or the HEAD is detached, "origin" is assumed to be the default remote. If the superproject doesn’t have a default remote configured the superproject is its own authoritative upstream and the current working directory is used instead.
```

In other words default-remote is:
 - remote of upstream branch of the current branch;
 - "origin" if origin is defined in the .git/config.
 
Possibly werf should detect remote of upstream branch of the current branch for the main worktree and tell git to use this default-remote when running git-submodule update on the service worktree with detached commit.

**Comment by alexey-igrychev** [comment]
This edge case (a non-`origin` remote combined with relative submodule URLs) was never addressed and hasn't been reported by users. Closing as not planned — please reopen with your git remote/submodule configuration if you hit it.
